### PR TITLE
Made annotation draw time configurable (Fixes #1, #15 and #28)

### LINKED ISSUE #1: Use beforeDraw to prevent overlapping
Consider using beforeDraw (or make it an option) so that the annotation doesn't overlap other chart elements. Currently the annotation gets drawn over tooltips. I can send through a pull request if you want.

image

image

### PR DESCRIPTION
**Pull request by petewalker**
Added a new option to the annotation plugin to allow configuration of
when the annotations should be drawn.
Documentation has been updated to reflect this, have also run `gulp
build`

### CODE REVIEW

**Comment by petewalker** [comment]
@etimberg any chance you could review?

File: Chart.Annotation.js [hunk 1]
```diff
@@ -111,11 +112,17 @@ var updateFunctions = Chart.Annotation.updateFunctions = {
...
+	afterDatasetsDraw: "afterDatasetsDraw",
+	beforeDatasetsDraw: "beforeDatasetsDraw",
+};
+
 // Chartjs Zoom Plugin
 var AnnotationPlugin = Chart.PluginBase.extend({
 	beforeInit: function(chartInstance) {
 		var options = chartInstance.options;
-		options.annotation = helpers.configMerge(options.annotation, Chart.Annotation.defaults);
+		options.annotation = helpers.configMerge(Chart.Annotation.defaults, options.annotation);
```
Comment (etimberg) [inline Chart.Annotation.js]: This is indeed more correct, can't believe I missed it before. Good catch!

**Comment by petewalker** [comment]
Hope this is good enough: https://jsfiddle.net/wfwLng73/

**Comment by petewalker** [comment]
(you can toggle the draw order in the select element at the bottom)

**Comment by etimberg** [comment]
Thanks, looks great!

**Comment by etimberg** [comment]
@petewalker I saw you had a bunch of PRs there, would you be interested in helping maintain the annotation plugin?

**Comment by petewalker** [comment]
@etimberg Possibly - I need it right now for a project I'm working on, so happy to continue pushing improvements.

**Comment by compwright** [comment]
@etimberg I would be interested in maintaining also.

**Pull request merged**